### PR TITLE
feat(mkdocs): implement versioning on site

### PR DESCRIPTION
**Pull request by cboin1996**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Purpose

Currently, there is no versioning on the documentation website for asea.

The work in my PR implements versioning, and adds a dark mode toggle for fun.

To view the site with versioning enabled, click [here](https://cboin1996.github.io/aws-secure-environment-accelerator/v1.5.6/), where I have deployed the pages site and updated the gh-pages branch on my fork to reflect the change.

We would also need to include a second pull request, which I have already created and linked [here](https://github.com/aws-samples/aws-secure-environment-accelerator/pull/1129).

TL;DR, this pull request adds this dropdown for versioned documentation of ASEA, and includes a workflow to auto-generate versions on new releases of ASEA:

image

## Notes

This change includes a modification to the `docs.yml` workflow. The workflow for updating the documentation site will
only be run on a release, since a tag is needed in order to version the site.

## Relevant Issue

Closes https://github.com/aws-samples/aws-secure-environment-accelerator/issues/1125

### CODE REVIEW

**Comment by cboin1996** [comment]
@dliggat 

For the failed test workflow, it looks like I forgot to add the below block to the “test” job

https://github.com/aws-samples/aws-secure-environment-accelerator/pull/1128/files#diff-9cf2000c53760d837a449f874e53f792819108d3a4bf346336d0f7d082deae2cL13

**Comment by cboin1996** [comment]
@dliggat, I have added the missing makefile which should allow the CI to test building the site prior to deployment.

https://github.com/aws-samples/aws-secure-environment-accelerator/pull/1128/commits/90b08084fe230f5100d50fc90c8f19823bbe62f3

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -1,37 +1,64 @@
...
     defaults:
       run:
         working-directory: ./src/mkdocs
+    if: ${{ github.event_name == 'release' }}
+    needs: [test]
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - name: set up Python 3.10
```
Comment (dliggat) [inline .github/workflows/docs.yml:42]: 3.10 vs 3.11

**Review by dliggat — APPROVED** [review]
LGTM

**Comment by cboin1996** [comment]
It looks like the new release `v1.5.6-a` has closed [my other PR](https://github.com/aws-samples/aws-secure-environment-accelerator/pull/1129) automatically. To finish this work, we will need to manually overwrite the `gh-pages` branch with what is contained in my branch: https://github.com/cboin1996/aws-secure-environment-accelerator/tree/gh-pages.

I also updated my gh-pages branch to contain my latest changes in this PR, so that they are included in the 1.5.6-a release.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (cboin1996) [reply .github/workflows/docs.yml]: Resolved here: https://github.com/aws-samples/aws-secure-environment-accelerator/pull/1128/commits/8903a553395033628cf97a9aeab33e476d98e72e
